src/core/bittorrent/session.cpp

```diff
@@ -1443,8 +1438,20 @@ void Session::setAppendLabelToSavePath(bool append)
 {
     if (m_appendLabelToSavePath != append) {
         m_appendLabelToSavePath = append;
-        foreach (TorrentHandlePrivate *const torrent, m_torrents)
-            torrent->handleDefaultSavePathChanged();
+        foreach (TorrentHandle *const torrent, m_torrents) {
+            QString label = torrent->label();
+            if (label.isEmpty()) continue;
+
+            QString testedOldSavePath = m_defaultSavePath;
+            QString newSavePath = m_defaultSavePath;
```

Comment:
(I have some down time and reviewing this on the go. Can't see whole source).
Do I understand correctly the below lines? They remove the label from the path for ALL torrents if the feature is disabled and they add it for ALL torrents if enabled, right?
In case we didn't discuss this change in #3495 then that's ok. I don't want you to recode this.
However, I wonder if other bt clients do the same when this feature is enabled/disabled or if they just manage the new torrents...